api/src/infrastructure/docker.rs

```diff
@@ -490,6 +531,73 @@ impl DockerInfrastructure {
         Ok(())
     }
 
+    async fn get_existing_volumes(app_name: &String) -> Result<Vec<VolumeInfo>, ShipLiftError> {
+        let docker = Docker::new();
+        docker.volumes().list().await.map(|volume_infos| {
+            volume_infos
+                .into_iter()
+                .filter(|s| {
+                    s.labels
+                        .as_ref()
+                        .map(|label| label.get(APP_NAME_LABEL) == Some(app_name))
+                        .unwrap_or(false)
+                })
+                .collect::<Vec<VolumeInfo>>()
+        })
+    }
+
+    async fn create_docker_volume(
+        app_name: &String,
```

Comment:
I suggest to run `cargo clippy` and see what lints are recommended for **new** code

```suggestion
        app_name: &str,
```